source/guidelines/authentication.rst

```diff
@@ -83,10 +83,76 @@ web sites and services; it also seamlessly interacts with
 documentation <https://www.keycloak.org/documentation>`_.
 
+.. note:: By `internal infrastructure` we mean also the access to Open
+   Data that are already available to the Open Data Hub Team but not
+   yet to the Data Consumers, and therefore require authentication.
+
 Source code for both the authentication server and a few pre-cooked
 examples of applications configured to connect to it can be found in
 dedicated servers created by the Open Data Hub Team: the
 `authentication server
 <https://github.com/noi-techpark/authentication-server-examples>`_,
 and the `example applications
 <https://github.com/noi-techpark/authentication-server-examples>`_,
+
+Quick howto
+~~~~~~~~~~~
+
+In order to access the internal infrastructure, you need first to get
+a token, then use it together with the API. Both steps can be achieved
+using command-line tools, for a programmatic access to the date, which
+is the method shown here.
+
+.. rubric:: Request an access token
+
+In order to receive an access token, you need in advance to have
+credentials for the Open Data Hub. If you do not have them, please
+open a ticket on issues.opendatahub.bz.it or send an email to
+:email:`help@opendatahub.bz.it`.
+
+With your username and password
+(:strong:`my_username`. `strong:`my_password`), the access token is
+granted to you with the following call:
+ 
+.. code-block:: bash
+   :name: grant-token
+   :caption: Receiving an access topic
+
+   curl -X POST -L -H 'Content-Type:application/x-www-form-urlencoded' \
+   "https://auth.opendatahub.bz.it/auth/realms/noi/protocol/openid-connect/token" \
+   -d 'grant_type=password&username=my_username&password=my_password&client_id=your-client-id'
+
+   
+The last parameter, :literal:`client_id` should be set to
+:strong:`odh-mobility-v2`.
+
+Since the token expires after a given amount of time, it might prove
+necessary to refresh it, an action that can be done by replacing the
+parameters given in the query above with
+
+.. code-block::
+   :name: refresh-token
+   :caption: Refreshing the access token
+	  
+   curl -X POST -L -H 'Content-Type:application/x-www-form-urlencoded' \
+   "https://auth.opendatahub.bz.it/auth/realms/noi/protocol/openid-connect/token" \
+   -d 'grant_type=refresh_token&refresh_token=*****&client_id=your-client-id'
+
+Here, use the refresh token received from
+:numref:`grant-token` and the same :literal:`client_id`. 
+
+
+.. rubric:: Retrieve data with the token.
+
+Once you received the access token, it is easy to use it in actual
+requests. The following API call shows how to get all
+:strong:`sname`\s and :strong:`mvalue`\s from the VMS dataset:
+	    
+.. code-block::
+   :name: get-closed-data
+   :caption: Retrieving data with the access token
+
+   curl -X GET 'https://mobility.api.opendatahub.bz.it/v2/api/flat/VMS/*?select=sname,mvalue' \
+   -H 'content-type: application/json' \
+   -H 'Authorization: bearer your-access-token'
+
```

Comment:
I am sorry @Piiit but... we (ODH) have decided that our message is "in Open Data Hub everything is by default open except for some (internal) stuff", why would we go the other way around and say that "this data is closed, but some is open"? And why you keep on asking me to add references to closed data vs open data? 

I will keep in the docs this approach and I will also discard any further comments on authentication and authorization as soon as this PR is closed, unless @ohnewein tells me otherwise.

Reply:
It is just an explanation, that some calls contain meta-data, and what that means. Since we write a authentication chapter, this is an important aspect, and has nothing todo with the discussions about open data. This chapter is hidden in the developers sections, so that is already fullfilled. It is not about open vs closed, but meta-data vs proper measurements.

Reply:
@Piiit, now that I read again your comment, I realise that in today's call you mentioned /latest but *not* /from-data/to-date  (btw: it should be from-dat**e**, not dat**A**, right?), are there other of these... errr... "methods"?

Reply:
@stefanodavid Yes, that is the other measurement giving call... there are no other calls. Just to be sure, we have:

- /v2
- /v2/apispec
- /v2/{representation}, which can be flat or tree
- /v2/{representation}/{stationtype}
- /v2/{representation}/{stationtype}/{datatype}
- /v2/{representation}/{stationtype}/{datatype}/latest
- /v2/{representation}/{stationtype}/{datatype}/from-date/to-date

Reply:
You can look at the swagger for more information

Reply:
> You can look at the swagger for more information, see https://mobility.api.opendatahub.bz.it

yes, of course, you're right... I still miss some detail of the "v2-version" of api v2 :-)

Reply:
@piit, this part will be covered in another issue, I think we can mark this as rewolved, so you can procced and merge the PR.